app/charts/shared/chart-helpers.tsx

```diff
@@ -160,4 +168,43 @@ export const getLabelWithUnit = (
     : dimension.label;
 };
 
+export const checkForMissingValuesInSegments = (
+  groupedMap: InternMap<string, Array<Observation>>,
```

Comment:
`observationsByX` would be clearer to me, I had trouble understanding by what the observations were grouped here.